lib/phoenix/controller.ex

```diff
@@ -535,6 +535,62 @@ defmodule Phoenix.Controller do
     end
   end
 
+  @doc """
+  Stores the root layout for rendering if one was not stored yet.
+
+  Raises `Plug.Conn.AlreadySentError` if `conn` is already sent.
+  """
+  @spec put_new_root_layout(Plug.Conn.t, {atom, binary | atom} | false) :: Plug.Conn.t
+  def put_new_root_layout(%Plug.Conn{state: state} = conn, layout)
+      when (is_tuple(layout) and tuple_size(layout) == 2) or layout == false do
+    if state in @unsent do
+      update_in conn.private, &Map.put_new(&1, :phoenix_root_layout, layout)
+    else
+      raise AlreadySentError
+    end
+  end
```

Comment:
I think we don't need this one because we don't have a default operation (like we call put_layout for controllers) that puts the root layout.